authentication-service/src/main/kotlin/com/saveourtool/save/authservice/config/WebSecurityConfig.kt

```diff
@@ -0,0 +1,130 @@
+/**
+ * Configuration beans for security in different profiles
+ */
+
+package com.saveourtool.save.authservice.config
+
+import com.saveourtool.save.authservice.security.ConvertingAuthenticationManager
+import com.saveourtool.save.authservice.security.CustomAuthenticationBasicConverter
+import com.saveourtool.save.authservice.utils.roleHierarchy
+import com.saveourtool.save.v1
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Profile
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.expression.method.DefaultMethodSecurityExpressionHandler
+import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
+import org.springframework.security.config.web.server.SecurityWebFiltersOrder
+import org.springframework.security.config.web.server.ServerHttpSecurity
+import org.springframework.security.crypto.factory.PasswordEncoderFactories
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.web.server.SecurityWebFilterChain
+import org.springframework.security.web.server.authentication.AuthenticationWebFilter
+import org.springframework.security.web.server.authentication.HttpStatusServerEntryPoint
+
+import javax.annotation.PostConstruct
+
+@EnableWebFluxSecurity
+@EnableReactiveMethodSecurity
+@Profile("secure")
+@Suppress("MISSING_KDOC_TOP_LEVEL", "MISSING_KDOC_CLASS_ELEMENTS", "MISSING_KDOC_ON_FUNCTION")
+class WebSecurityConfig(
+    private val authenticationManager: ConvertingAuthenticationManager,
```

Comment:
suggestion is to remove a dependency to profile in shared library and use `Conditional`:
- `WebSecurityConfig` can have conditional on `ConvertingAuthenticationManager`
- `NoopWebSecurityConfig` can have conditional on missing `SecurityWebFilterChain`

Reply:
Tried to use `@ConditionalOnBean` but it leads to 403 Forbidden